src/test/java/dk/cngroup/lentils/service/HintServiceTest.java

```diff
@@ -0,0 +1,76 @@
+package dk.cngroup.lentils.service;
+
+import dk.cngroup.lentils.LentilsApplication;
+import dk.cngroup.lentils.entity.Cypher;
+import dk.cngroup.lentils.entity.Hint;
+import dk.cngroup.lentils.repository.HintRepository;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {LentilsApplication.class, ObjectGenerator.class})
+public class HintServiceTest {
+
+    @InjectMocks
+    HintService hintService;
+
+    @Mock
+    HintRepository hintRepository;
+
+    @Autowired
+    private CypherService cypherService;
+
+    @Autowired
+    private ObjectGenerator generator;
+
+    private List<Cypher> cyphers;
+
+    @Before
+    public void setUp() throws Exception {
+        cypherService.saveAll(generator.generateCypherList());
+        cyphers = cypherService.getAll();
+    }
+
+    @Test
+    public void getHintsForCypher() {
+        Cypher cypher = getTestedCypher();
+
+        List<Hint> hints = generator.generateHintList(cyphers);
+        when(hintRepository.saveAll(hints)).thenReturn(hints);
+        hintService.addAll(hints);
+
+//        when(hintRepository.findByCypher(cypher))
+//        List<Hint> hintsForCypher = hintService.getHintsForCypher(cypher);
+
+    }
+
+    @Test
+    public void addHintForCypher() {
+        Cypher cypher = getTestedCypher();
+        Hint hint = new Hint("hint", 5, cypher);
+
+        when(hintRepository.save(hint)).thenReturn(hint);
+
+        assertEquals(hint, hintService.add(hint));
+    }
+
+    private Cypher getTestedCypher() {
```

Comment:
Tuto magiskou strukturu bych taky odstranil. Je to zbytecne komplikovane. Kdys vis, ze testujes jen cyper kde stage=3, tak proc nepracujes jen primo se sifrou, ktera ma tu stage=3? To je mnohem lepsi. Nehlede na to, ze to TESTED_STAGE se nachazi v uplne jinem souboru, nez samotny test. Vsechny parametry, konfigurace atd by se meli nachazet primo v tom testu aby bylo hned videt s cim se pracuje. Do toho ObjectGeneratoru je pak muzes poslat parametrem, ale proc by mel ObjectGenerator obsahovat informaci, ktera konkretni sifra se testuje? Co kdyz budes chtit najednou otestovat sifru c. 4? Pridas konstantu do ObjectGeneratoru s nazvem `TESTED_STAGE_2`? Nebo tu druhou uz nechas v tom novem testu, ktery bude testovat tuto sifru cislo 4? <- Tyto recnicke otazky pisu proto, abych te navedl, ke spravnemu reseni a aby si si sama uvedomila mozne problemy s existujicim reseni a aby si si vsimla, ze je lepsi, kdyz konfigurace, hodnoty, atd, ktere testujes v nejakem testu byly ulozeny primo v nem a nikde jinde ;)